01/most-frequent-k-elements.md

```diff
@@ -0,0 +1,172 @@
+4. Must the output list be in a particular order?
+
+
+##### !end-explanation
+
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 2 -->
+<!-- prettier-ignore-start -->
+
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: afcb3fb9-b225-4d48-b255-b0f3508de237
+* title: Write Unit Tests
+* topics: pse
+##### !question
+
+1. Use the comments provided to write at least two example input/outputs:
+    * Consider at least one nominal and one edge case.
+    * What is the expected output for the given input?
+    * You can use the examples provided in the prompt, or other examples.
+2. Write unit tests for `most_frequent_k_elements` for the nominal and edge cases you identified in the first step.
+
+*Note: Click the **Run Tests** button to save your tests for instructor feedback. No real tests are actually run again your unit tests.*
+
+##### !end-question
+##### !placeholder
+
+```py
+# example input 1:
+# expected output 1:
+
+# example input 2:
+# expected output 2:
+
+def test_nominal_case():
+    # ^rename with meaningful test name
+    # and complete test implementation below
+    pass
+    # arrange
+
+    # act
+
+    # assert
+
+def test_edge_case():
+    # ^rename with meaningful test name
+    # and complete test implementation below
+    pass
+    
+    # arrange
+    
+    # act
+    
+    # assert
+```
+##### !end-placeholder
+
+##### !tests
+
+```py
+import unittest
+
+class TestPython1(unittest.TestCase):
+  def test_always_pass(self):
+    self.assertEqual(1,1)
+```
+
+##### !end-tests
+##### !explanation 
+
+Example tests:
+
+```python
+def test_negative_numbers(arr,k):
+    # Arrange
+    arr = [-1,-1,2,3,3,3]
+    k = 2 
+    # Act
+    result = most_frequent_k_elements(arr, k)
+
+    # Assert
+    assert result == [3,-1]
+
+def test_unordered_numbers():
+    # Arrange
+    arr = [3, 1, 2, 3, 1, 3]
+    k = 2 
+    # Act
+    result = most_frequent_k_elements(arr, k)
+
+    # Assert
+    assert result == [3, 1]
+```
+
+##### !end-explanation
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- Question 3 -->
+<!-- prettier-ignore-start -->
+### !challenge
+* type: paragraph
+* id: 2df04c34-94fb-45f3-869a-09f96fef5e6b
+* title: Create Logical Steps
+* topics: pse
+##### !question
+
+Without writing code, describe how you would implement `most k frequent elements` using hash tables in enough detail that someone else could write the code. 
+* It may be helpful to break up the problem/algorithm into smaller subproblems/algorithms. For example, 1. Handle invalid input, 2. Given valid input, perform the computation/solve the problem/etc.
+* Your logical steps could take the form of a numbered list, pseudo code, or anywhere in between. What's important at this stage is to think through and outline the implementation before writing code.
+
+##### !end-question
+
+##### !placeholder
+
+Write the logical steps here.
+
+##### !end-placeholder
+
+##### !hint
```

Comment:
I like these hints, but can the original steps you had in the previous commit come back as the explanation?